UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Helpers/NMHelper.cs

```diff
@@ -10,19 +10,60 @@ public static class NMHelper
     {
         public static List<ShowFilesResponseModel> ListFilesResponse(BatchSearchResponse SearchResult)
         {
+            List<BatchDetails> batchDetailsList = new();
             if (SearchResult.Entries.Count > 1)
             {
-                List<BatchDetails> batchDetailsList = new();
+                List<BatchDetails> distributorBatchDetail = new();
+                List<BatchDetails> publicBatchDetail = new();
+                for (int i = 0; i < SearchResult.Entries.Count; i++)
+                {
+                    bool checkContent = false;
+                    for (int j = 0; j < SearchResult.Entries[i].Attributes.Count; j++)
+                    {
+                        if (SearchResult.Entries[i].Attributes[j].Value == "tracings")
+                        {
+                            checkContent = true;
+                        }
+                    }
+                    if (checkContent)
+                    {
+                        distributorBatchDetail.Add(AddBatchDetails(SearchResult, i));
+                    }
+                    else
+                    {
+                        publicBatchDetail.Add(AddBatchDetails(SearchResult, i));
+                    }
+                }
```

Comment:
This can be refactored as below:
```suggestion
               for (int i = 0; i < SearchResult.Entries.Count; i++)
                {
                    if (SearchResult.Entries[i].Attributes.SingleOrDefault(x => x.Key.Contains("Content") && x.Value.Contains("tracings")) != null)
                    {
                            distributorBatchDetail.Add(AddBatchDetails(SearchResult, i));                                               
                    }
                    else
                    {
                        publicBatchDetail.Add(AddBatchDetails(SearchResult, i));
                    }
              }
```